truffle/com.oracle.truffle.api.object/src/com/oracle/truffle/api/object/Layout.java

```diff
@@ -65,11 +65,30 @@ public static Layout createLayout() {
 
     public abstract Class<? extends DynamicObject> getType();
 
-    public abstract Shape createShape(ObjectType operations);
+    /**
+     * Create a root shape.
+     *
+     * @param objectType that describes the object instance with this shape.
+     */
+    public abstract Shape createShape(ObjectType objectType);
 
-    public abstract Shape createShape(ObjectType operations, Object sharedData);
+    /**
+     * Create a root shape.
+     *
+     * @param objectType that describes the object instance with this shape.
+     * @param sharedData for language-specific use
+     */
+    public abstract Shape createShape(ObjectType objectType, Object sharedData);
 
-    public abstract Shape createShape(ObjectType operations, Object sharedData, int id);
```

Comment:
@andreas (no github account yet) Can you create something else than a root shape manually? Maybe we should name the method also createRootShape.